api/middleware/synchronize.js

```diff
@@ -0,0 +1,66 @@
+const logger = require('../logger')('sync middleware');
+const { cache, modelIndex } = require('./cache');
+
+const synchronizeAll = getSpecifics =>
+  cache(
+    ['sync-all', modelIndex(getSpecifics)],
+    () => async (req, res, next) => {
+      let action;
+
+      try {
+        const { modelClass, foreignKey, action: actionToReport } = getSpecifics(
```

Comment:
Naming things is hard.  `getSpecifics` seems dumb.  Maybe `getDetails`?  I dunno...